infra/notifications/src/app.js

```diff
@@ -165,68 +165,107 @@ app.post('/', async (req, res) => {
 
 /**
  * Endpoint called from the mobile marketplace app to register a device token
- * and Ethereum address.
+ * and Ethereum address for push notifications.
+ *
+ * Notes:
+ *  - The app may call this end point more than once.
+ *  - On iOS devices, the app may call this endpoint before the user accepts
+ *    to turn on notifications. So the deviceToken argument may be empty.
  */
 app.post('/mobile/register', async (req, res) => {
-  logger.info('Call to mobile device registry endpoint')
-
   const mobileRegister = {
-    ethAddress: req.body.eth_address,
-    deviceType: req.body.device_type,
+    ethAddress: _.get(req.body, 'eth_address', null),
+    deviceType: _.get(req.body, 'device_type', null),
     deviceToken: _.get(req.body, 'device_token', null),
-    permissions: req.body.permissions
+    permissions: _.get(req.body, 'permissions', null)
   }
+  logger.info(`POST /mobile/register for ${mobileRegister.ethAddress}`)
 
-  // See if a row already exists for this device/address
-  let registryRow = await MobileRegistry.findOne({
-    where: {
-      ethAddress: mobileRegister.ethAddress,
-      deviceToken: mobileRegister.deviceToken
+  if (!mobileRegister.ethAddress) {
+    return res.status(400).send({ errors: ['Invalid Eth address'] })
```

Comment:
Maybe sometime later we clean this up with the same validation library we use in t3, not important now but would be nice

Reply:
Yes good idea. Would be nice to use that express-validator library everywhere. I added a TODO.